src/internal/modules/invoice-accounts/controllers/invoice-accounts.js

```diff
@@ -70,18 +98,49 @@ const getAddress = async (request, h) => {
 
 const postAddress = async (request, h) => {
   const { regionId, companyId } = request.params;
-  const addresses = await dataService.getCompanyAddresses(companyId);
+  const session = dataService.sessionManager(request, regionId, companyId);
+  const addresses = await dataService.getCompanyAddresses(companyId, session);
   const schema = selectAddressFormSchema(request.payload);
   const form = forms.handleRequest(selectAddressForm(request, addresses), request, schema);
   if (form.isValid) {
     const { selectedAddress } = forms.getValues(form);
-    dataService.sessionManager(request, regionId, companyId, { address: { addressId: selectedAddress } });
+    if (selectedAddress !== tempId) {
+      dataService.sessionManager(request, regionId, companyId, { address: { id: selectedAddress } });
+    }
     const redirectPath = selectedAddress === 'new_address' ? 'create-address' : 'add-fao';
     return h.redirect(`/invoice-accounts/create/${regionId}/${companyId}/${redirectPath}`);
   }
   return h.postRedirectGet(form, urlJoin('/invoice-accounts/create/', regionId, companyId, 'select-address'));
 };
 
+const contactEntryHandover = async (request, h) => {
```

Comment:
we start route controller methods with the method type .i.e get post patch delete so this should be getContactEntryHandover. This is to avoid confusion over other helper methods that is also in the controller but not directly linked to a route.